site/_posts/2020-07-02-auth-and-the-bottleneck-architecture.md

```diff
@@ -0,0 +1,83 @@
+
+If your auth system has mature user management functionality, it will accelerate custom application development by providing often used services. No more worrying about building a front end to allow customer service reps to lock accounts. Nor do you have to build a way to let end users reset their passwords or register.
+
+### Operational Ease
+
+Such a centralized user auth service also makes it easy to turn accounts on and off. Onboarding a new employee becomes simpler, and offboarding departing employees is a matter of disabling their account in one place, rather than hunting down all the applications to which they have access. Or worse, leaving those accounts enabled.
+
+There are additional benefits to end users, as well. While identifying people with username, password and multi-factor authentication is a secure and relatively convenient method, there may be times when an alternative is a better user experience.  
+
+
+### Give Your Users the Experience They Want
+
+As you might expect, different types of users have different third party accounts. If you have a consumer focused application, offering sign on with Facebook is a great idea, because everyone has a Facebook account. One less password for your potential users to remember, and one less obstacle to signing up. 
+
+On the other hand, if the application is aimed at enterprise customers, integrating with ActiveDirectory can help adoption. If developers are your target market, GitHub authentication eases the sign-up process as well as signals to them that you understand their needs.
+
+If your auth system allows for multiple different identity providers, your application can meet users "where they are", rather than requiring them to create and manage a new account. 
+
+If you work for a large organization, you may need to federate your user management system, using one of the standards such as OIDC, to other user datastores. Having a central service for your suite of applications means federation and integration only needs to be done once, rather than for each application you build.
+
+## Challenges
+
+Of course, nothing is perfect. There are challenges with this approach as well. Some technical, some not so much. 
+
+First off is the [SSO tax](https://sso.tax/). Many third party applications don't support this user auth delegation until you are on an enterprise plan. Ouch. Investigate required applications to see if and how they can act as a RP or SP to a centralized user management system before you decide to pursue this architecture. 
+
+Another organizational challenge is ensuring developers and users don't do an end run around the organization's user management system. Encourage employees to work within these constraints by making it as easy as possible and repeatedly explaining the benefits. Providing examples of successful integrations can help with both goals.
```

Comment:
I had to look up "end run around" and I think I get it. Maybe something a little more approachable?

Also, the first sentence says "developers and users", but this later talks about employees. Are "users" end-users or employees?

Reply:
Clarified.